src/sourcemapUtils.ts

```diff
@@ -0,0 +1,117 @@
+// Copy of https://github.com/react-native-community/cli/blob/13.x/packages/cli-hermes/src/profileHermes/metroBundleOptions.ts as `cli-hermes` was recently removed from React Native Community CLI
+
+import fs from 'fs';
+import path from 'path';
+import os from 'os';
+import type { SourceMap } from 'hermes-profile-transformer';
+import type { MetroBundleOptions } from './getMetroBundleOptions';
+
+type Config = any;
+
+function getTempFilePath(filename: string) {
+  return path.join(os.tmpdir(), filename);
+}
+
+function writeJsonSync(targetPath: string, data: any) {
+  let json;
+  try {
+    json = JSON.stringify(data);
+  } catch (e) {
+    throw new Error(
+      `Failed to serialize data to json before writing to ${targetPath}`,
+      e as Error
+    );
+  }
+
+  try {
+    fs.writeFileSync(targetPath, json, 'utf-8');
+  } catch (e) {
+    throw new Error(`Failed to write json to ${targetPath}`, e as Error);
+  }
+}
+
+async function getSourcemapFromServer(
+  port: string,
+  { platform, dev, minify, host }: MetroBundleOptions
+): Promise<SourceMap | undefined> {
+  console.log('Getting source maps from Metro packager server');
+
+  const requestURL = `http://${host}:${port}/index.map?platform=${platform}&dev=${dev}&minify=${minify}`;
+  console.log(`Downloading from ${requestURL}`);
```

Comment:
as written above, don't expect the `host` to be passed but rather get the Ip address here like it was done before in the hermes-cli package